### PR TITLE
Chilly Reception AI

### PR DESCRIPTION
**Pull request by kittenchilly**
## Description
Adds AI behavior for chilly reception, acting much like a combination of how the ai uses parting shot and snowscape.

## **Discord contact info**
kittenchilly

### CODE REVIEW

**Review by Pawkkie — COMMENTED** [review]
I think this looks great as is, really clean way of implementing the behaviour. The only thing I could consider adding is a test to make sure the intended behaviour of not looping between snow and hail works properly, but I don't feel that's necessary / blocking.

I'd prefer another set of eyes before accepting as I'm still very new to reviewing PRs :)

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -1739,6 +1739,15 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
             break;
         case EFFECT_HIT_ESCAPE:
             break;
+        case EFFECT_CHILLY_RECEPTION:
+            if (CountUsablePartyMons(battlerAtk) == 0)
+                ADJUST_SCORE(-10);
+            else if (weather & (B_WEATHER_SNOW | B_WEATHER_PRIMAL_ANY)
+             || IsMoveEffectWeather(aiData->partnerMove))
```
Comment (AlexOn1ine) [inline src/battle_ai_main.c:1746]:
```suggestion
            else if (weather & (B_WEATHER_SNOW | B_WEATHER_PRIMAL_ANY) || IsMoveEffectWeather(aiData->partnerMove))
```

Comment (Pawkkie) [reply src/battle_ai_main.c:1746]: For my own knowledge for future reviewing, is there a threshold beyond which we prefer multiline conditions vs single line conditionals?

Comment (AlexOn1ine) [reply src/battle_ai_main.c:1746]: 2 conditions can for sure go on one line, after that it is up to reviewer.

Comment (kittenchilly) [reply src/battle_ai_main.c:1746]: How I did it is consistent with Snowscape.

Comment (AlexOn1ine) [reply src/battle_ai_main.c:1746]: You can change that as well if you want but how Snowscape didn't shouldn't generally matter here. It is really just a nitpick :sweat_smile:

Comment (kittenchilly) [reply src/battle_ai_main.c]: Ok, done

**Review by Pawkkie — APPROVED** [review]

**Pull request merged**